spec.html

```diff
@@ -13524,70 +13526,58 @@ <h1>
         1. Else,
-          1. Perform ? IteratorBindingInitialization of _formals_ with arguments _iteratorRecord_ and _env_.
-        1. If _hasParameterExpressions_ is *false*, then
-          1. NOTE: Only a single Environment Record is needed for the parameters and top-level vars.
-          1. Let _instantiatedVarNames_ be a copy of the List _parameterBindings_.
-          1. For each element _n_ of _varNames_, do
-            1. If _n_ is not an element of _instantiatedVarNames_, then
-              1. Append _n_ to _instantiatedVarNames_.
-              1. Perform ! _env_.CreateMutableBinding(_n_, *false*).
-              1. Perform ! _env_.InitializeBinding(_n_, *undefined*).
-          1. Let _varEnv_ be _env_.
-        1. Else,
-          1. NOTE: A separate Environment Record is needed to ensure that closures created by expressions in the formal parameter list do not have visibility of declarations in the function body.
+          1. Perform ? IteratorBindingInitialization of _formals_ with arguments _argumentsIterator_ and _env_.
+        1. NOTE: If the formal parameter list contains any expressions, a separate Environment Record for the function body is needed to ensure that declarations in the function body are not visible to closures created by those expressions.
+        1. If _hasParameterExpressions_ is *true*, then
           1. Let _varEnv_ be NewDeclarativeEnvironment(_env_).
           1. Set the VariableEnvironment of _calleeContext_ to _varEnv_.
           1. Let _instantiatedVarNames_ be a new empty List.
-          1. For each element _n_ of _varNames_, do
-            1. If _n_ is not an element of _instantiatedVarNames_, then
-              1. Append _n_ to _instantiatedVarNames_.
-              1. Perform ! _varEnv_.CreateMutableBinding(_n_, *false*).
-              1. If _n_ is not an element of _parameterBindings_ or if _n_ is an element of _functionNames_, let _initialValue_ be *undefined*.
-              1. Else,
-                1. Let _initialValue_ be ! _env_.GetBindingValue(_n_, *false*).
-              1. Perform ! _varEnv_.InitializeBinding(_n_, _initialValue_).
-              1. NOTE: A var with the same name as a formal parameter initially has the same value as the corresponding initialized parameter.
+        1. Else,
+          1. Let _varEnv_ be _env_.
+          1. Let _instantiatedVarNames_ be a copy of the List _parameterBindings_.
+        1. For each element _n_ of _varNames_, do
+          1. If _n_ is not an element of _instantiatedVarNames_, then
+            1. Append _n_ to _instantiatedVarNames_.
+            1. Perform ! _varEnv_.CreateMutableBinding(_n_, *false*).
+            1. NOTE: A var with the same name as a formal parameter initially has the same value as the corresponding initialized parameter.
```

Comment:
This refactoring does preserve observable behavior, from what I can tell, but it also kind of obscures what's going on here. I think this note should explain what was previously implicit in the structure. Something like

```suggestion
            1. NOTE: When there are expressions in the parameter list, it is possible to observe a difference between a variable in the parameter list and a variable in the body with the same name. The variable in the body initially has the same value as the corresponding initialized parameter.
```

Though, I do think eventually we will want to drop the whole `hasParameterExpressions` thing as mentioned in https://github.com/tc39/ecma262/issues/623, and instead have a NOTE pointing out the possibility of optimizing out the distinction between the two variables when that distinction is not observable. This refactoring makes it easier to get there, so :+1: